docs/design_docs/shared_steps.md

```diff
@@ -0,0 +1,74 @@
+# Shared steps
+
+date: 2023/08/18
+
+Contributors: Carolyn Begeman, Xylar Asay-Davis
+
+## Summary
+
+The capability designed here is the ability to share steps across test cases.
+In this design document, "shared steps" refers to any step which may be used by
+multiple test cases that are available in polaris.
+
+The main motivation behind this capability is the computational expense of
+running steps that could shared across test cases multiple times. In order to
+reflect the fact that steps are shared to the user, we present a new design for
+the working directory structure. The design is successful insofar as it
+guarantees that shared steps are run once per slurm job and that the role of
+shared steps is clear to users.
+
+## Requirements
+
+### Desired: Shared steps are set up once.
+
+Ideally, steps used by multiple test cases should be run once per invocation of
+`polaris setup` or `polaris suite`. This requirement may be dropped since step
+set up is not generally resource intensive.
+
+### Requirement: Shared steps are run once.
+
+Shared steps should be run once per invocation of `polaris serial` or
+`polaris parallel`.
+
+### Requirement: Shared steps are run before steps that depend on their output.
+
+### Requirement: Object properties of shared steps are intuitive.
```

Comment:
I'm not quite sure what "Object properties" means.  Does this refer to "attributes" of the class?